### ISSUE TITLE
Add "Smart break line" feature to LSP

### ISSUE THREAD

**Issue by DoctorKrolic** [microsoft/language-server-protocol#1586]
Visual Studio (not Code) has a great feature called "Smart break line" for C#. When you have code like this:
```cs
void M($$)
```
... pressing `Shift` + `Enter` turns it into this:
```cs
void M()
{
    $$
}
```
(`$$` represents cursor position)
Such feature can come in handy in a large variety of languages, so I think it should be added to LSP.

**Possible use cases:**
C++:
```cpp
void main($$)
```
```cpp
void main()
{
    $$
}
```
Python:
```py
def main($$)
```
```py
def main():
    $$
```
Html case 1:
```html
<div$$>
```
```html
<div>
    $$
</div>
```
Html case 2:
```html
<div id="someId$$">
```
```html
<div id="someId">
    $$
</div>
```
SQl (postgres dialect):
```sql
CREATE TABLE "MyTable$$"
```
```sql
CREATE TABLE "MyTable"
(
    $$
)
```

**Comment by rwols** [comment]
Same as https://github.com/microsoft/language-server-protocol/issues/1001 ?

**Comment by DoctorKrolic** [comment]
No, they are not the same. https://github.com/microsoft/language-server-protocol/issues/1001 is just about having some basic changes on pressing `Enter`, while here I don't want my cursor to jump and create parentheses/closing tag/whatever on every `Enter` I press. Here I as a user want to consciously controll when the feature is executed. `Shift` + `Enter` is just a key combination for it in VS, different editors might have different hotkeys for it
